core/src/main/scala/chisel3/Bits.scala

```diff
@@ -148,6 +148,21 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends Element wi
 
   /** @group SourceInfoTransformMacro */
   final def do_extract(x: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = {
+    this.widthOption.foreach { thisWidth =>
+      if (thisWidth == 0) {
+        Builder.error(s"Cannot extract from zero-width")
+      } else if (thisWidth == 1) {
+        Builder.warning("Extracting from a one-width element")
+      } else {
+        x.widthOption.foreach { xWidth =>
+          if (xWidth >= 31 || (1 << (xWidth - 1)) >= thisWidth) {
+            Builder.warning(s"Dynamic index with width $xWidth is too large for extractee of width $thisWidth")
```

Comment:
```suggestion
            Builder.warning(s"Dynamic index '$x' is too large for extractee '$this'")
```
This changes the typical warning from
```
[warn] src/test/scala/chiselTests/UIntOps.scala:390:16: Dynamic index with width 2 is too large for extractee of width 2
```
To
```
[warn] src/test/scala/chiselTests/UIntOps.scala:390:16: Dynamic index 'TooWide.index: IO[UInt<2>]' is too large for extractee 'TooWide.in: IO[UInt<2>]'
```